src/networkinfo.js

```diff
@@ -24,6 +24,14 @@ function NetworkInfo() {
         return this.options[support_name.toUpperCase()];
     };
 
+    this.isChannelName = function isChannelName(channel_name) {
+        if (typeof channel_name !== 'string' || channel_name === '') {
+            throw new TypeError('channel_name must be a non-empty string');
```

Comment:
rather than throwing an error, simply returning false here would work better as those values are obviously not channel names

Reply:
My thinking was that if these conditions are true, there is definitely a bug in the code that called the method. So throwing seemed more appropriate: fail loudly instead of silently when something doesn't make sense.

Specifically, I was worried it would be easy to mistakenly pass an IrcChannel object instead of a channel name string since we aren't using typescript/flow. Then the caller would get `false` back when really what they meant to do would have returned `true`.

But I'm not sure what the consequences would be if an exception is thrown and not handled, so I will defer to you on this if you still don't like the idea of throwing.

Reply:
Though throwing an error would be more specific, we don't throw errors anywhere else in the project so this should stay in line with that.

Reply:
Fair enough. Changed it. I also withdrew the additional commit updating `package-lock.json` as I just discovered `npm` has reintroduced another bug that causes the lockfile to be updated every time `npm install` runs.